factoring/interfaces.py

```diff
@@ -16,17 +16,34 @@
 
 import time
 import logging
-
+import functools
 from collections import OrderedDict
 
 import dwavebinarycsp as dbc
 from dwave.system.samplers import DWaveSampler
+from dwave.cloud.exceptions import SolverOfflineError
 import minorminer
 import dimod
 
 log = logging.getLogger(__name__)
 
 
+def qpu_ha(f):
+    """High-availability QPU wrapper: retry (with next solver) if active solver
+    goes offline.
+    """
+
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        while True:
```

Comment:
Do you not want to limit the number of attempts here?

Reply:
I meant, don't you want to stop and return an error if there are no online solvers?

Reply:
Generic `SolverError` is passed thru.